ssl verification fails on Windows

The s3.get module fails on Windows with an SSL error that seems to be related to a trust issue and/or certs not being available.

Command: `salt-call s3.get mybucket mykey/myfile.zip`
Versions report:
```
           Salt: 2014.7.1
         Python: 2.7.5 (default, May 15 2013, 22:44:16) [MSC v.1500 64 bit (AMD64)]
         Jinja2: 2.7.1
       M2Crypto: 0.21.1
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.1.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: Not Installed
```

Log output:
```
[INFO    ] Starting new HTTP connection (1): 169.254.169.254
[DEBUG   ] "GET /latest/meta-data/iam/security-credentials/ HTTP/1.1" 200 24
[INFO    ] Starting new HTTP connection (1): 169.254.169.254
[DEBUG   ] "GET /latest/meta-data/iam/security-credentials/my-role HTTP/1.1" 200 930
[DEBUG   ] CanonicalizedResource: /mybucket/mykey/myfile.zip
[DEBUG   ] x-amz-security-token
[DEBUG   ] CanonicalizedAmzHeaders: x-amz-security-token:SNIP
[DEBUG   ] String To Sign::
GET

text/plain
Thu, 22 Jan 2015 21:10:24 GMT
x-amz-security-token:SNIP
/mybucket/mykey/myfile.zip
[DEBUG   ] S3 Request: https://mybucket.s3.amazonaws.com/mykey/myfile.zip
[DEBUG   ] S3 Headers::
[DEBUG   ]     Authorization: AWS SNIP
[INFO    ] Starting new HTTPS connection (1): mybucket.s3.amazonaws.com
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
SSLError: [Errno 185090050] _ssl.c:340: error:0B084002:x509 certificate routines:X509_load_cert_crl_file:system lib
Traceback (most recent call last):
  File "<string>", line 6, in <module>
  File "__main__.py", line 726, in <module>
  File "__main__.py", line 332, in bootstrap
  File "__main__.py", line 359, in chainload
  File "__main__.py", line 715, in _chainload
  File "__main__.py", line 128, in <module>
  File "__main__salt-call__.py", line 14, in <module>
  File "salt/scripts.py", line 123, in salt_call
  File "salt/cli/__init__.py", line 420, in run
  File "salt/cli/caller.py", line 225, in run
  File "salt/cli/caller.py", line 129, in call
  File "salt/modules/s3.py", line 154, in get
  File "salt/utils/s3.py", line 155, in query
  File "requests/api.py", line 44, in request
  File "requests/sessions.py", line 456, in request
  File "requests/sessions.py", line 559, in send
  File "requests/adapters.py", line 382, in send
SSLError: [Errno 185090050] _ssl.c:340: error:0B084002:x509 certificate routines:X509_load_cert_crl_file:system lib
Traceback (most recent call last):
  File "<string>", line 6, in <module>
  File "__main__.py", line 726, in <module>
  File "__main__.py", line 332, in bootstrap
  File "__main__.py", line 359, in chainload
  File "__main__.py", line 715, in _chainload
  File "__main__.py", line 128, in <module>
  File "__main__salt-call__.py", line 14, in <module>
  File "salt/scripts.py", line 123, in salt_call
  File "salt/cli/__init__.py", line 420, in run
  File "salt/cli/caller.py", line 225, in run
  File "salt/cli/caller.py", line 129, in call
  File "salt/modules/s3.py", line 154, in get
  File "salt/utils/s3.py", line 155, in query
  File "requests/api.py", line 44, in request
  File "requests/sessions.py", line 456, in request
  File "requests/sessions.py", line 559, in send
  File "requests/adapters.py", line 382, in send
requests.exceptions.SSLError: [Errno 185090050] _ssl.c:340: error:0B084002:x509certificate routines:X509_load_cert_crl_file:system lib
```

This is true for me too, I have win_repo packages with installers located on Azure storage (https). They worked fine previously but on the latest version of the minion (2014.7.0) on Windows 8.1 I get the same error.

```
----------
          ID: 7zip
    Function: pkg.installed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "salt/state.py", line 1533, in call
                File "salt/states/pkg.py", line 831, in installed
                File "salt/modules/win_pkg.py", line 558, in install
                File "salt/modules/cp.py", line 328, in cache_file
                File "salt/fileclient.py", line 149, in cache_file
                File "salt/fileclient.py", line 591, in get_url
                File "requests/api.py", line 55, in get
                File "requests/api.py", line 44, in request
                File "requests/sessions.py", line 456, in request
                File "requests/sessions.py", line 559, in send
                File "requests/adapters.py", line 382, in send
              SSLError: [Errno 185090050] _ssl.c:340: error:0B084002:x509 certificate routines:X509_load_cert_crl_file:system lib
     Started: 14:06:44.730000
    Duration: 2744.0 ms
     Changes:
```

That doesn't work - I would guess the C API portion can't hit the pem that lives inside the zip file.

If I extract the requests directory from the zip file and drop it in `C:\salt\salt-2014.7.1.win-amd64\requests`, then put the cacert.pem file in there, the error changes: `requests.exceptions.SSLError: [Errno 1] _ssl.c:504: error:14090086:SSL routines:
SSL3_GET_SERVER_CERTIFICATE:certificate verify failed`

It's at least finding the file then, but it has to be on the filesystem, not wrapped in the library.zip file.

--- a/salt/utils/http.py
+++ b/salt/utils/http.py
@@ -59,6 +59,12 @@ try:
 except ImportError:
     HAS_MSGPACK = False
 
+try:
+    import certifi
+    HAS_CERTIFI = True
+except ImportError:
+    HAS_CERTIFI = False
+
 log = logging.getLogger(__name__)
 JARFILE = os.path.join(syspaths.CACHE_DIR, 'cookies.txt')
 SESSIONJARFILE = os.path.join(syspaths.CACHE_DIR, 'cookies.session.p')
@@ -454,6 +460,9 @@ def get_ca_bundle(opts=None):
     ):
         if os.path.exists(path):
             return path
+
+    if salt.utils.is_windows():
+        return certifi.where()
 
     return None
 

--- a/salt/utils/http.py
+++ b/salt/utils/http.py
@@ -461,7 +461,7 @@ def get_ca_bundle(opts=None):
         if os.path.exists(path):
             return path
 
-    if salt.utils.is_windows():
+    if salt.utils.is_windows() and HAS_CERTIFI:
         return certifi.where()
 
     return None

